client/events.lua

```diff
@@ -208,3 +194,30 @@ RegisterNetEvent('QBCore:Client:OnSharedUpdateMultiple', function(tableName, val
     end
     TriggerEvent('QBCore:Client:UpdateObject')
 end)
+
+-- Set vehicle props
+---@param vehicle number
+---@param props table<any, any>
+qbx.entityStateHandler('setVehicleProperties', function(vehicle, _, props)
+    if not props then return end
+
+    SetTimeout(0, function()
+        local state = Entity(vehicle).state
+
+        while state.setVehicleProperties do
+            if NetworkGetEntityOwner(vehicle) == cache.playerId then
+                for i = -1, 0 do
+                    local ped = GetPedInVehicleSeat(vehicle, i)
+                    if ped ~= cache.ped and ped > 0 then
+                        DeleteEntity(ped)
+                    end
+                end
```

Comment:
This is very flawed, just because you are the entity owner of the car dosnt mean u are the entity owner of the ped and therefor cant delete it.

Also shouldnt be used in a loop